src/modules/iotjs_module_websocket.c

```diff
@@ -110,14 +110,25 @@ static char *iotjs_ws_write_data(char *buff, void *data, size_t size) {
 }
 
 
-static bool iotjs_check_handshake_key(char *server_key) {
+static bool iotjs_check_handshake_key(char *server_key, jerry_value_t jsref) {
+  void *native_p;
+  JNativeInfoType *out_native_info;
+  bool has_p =
+      jerry_get_object_native_pointer(jsref, &native_p, &out_native_info);
+  if (!has_p || out_native_info != &wsclient_native_info) {
+    return false;
+  }
+  iotjs_wsclient_t *wsclient = (iotjs_wsclient_t *)native_p;
+
   unsigned char *out_buff = NULL;
 
-  size_t concatenated_size = strlen(WS_GUID) + strlen((char *)generated_key);
+  size_t concatenated_size =
+      strlen(WS_GUID) + strlen((char *)wsclient->generated_key);
   unsigned char concatenated[concatenated_size + 1];
 
-  memcpy(concatenated, generated_key, strlen((char *)generated_key));
-  memcpy(concatenated + strlen((char *)generated_key), WS_GUID,
+  memcpy(concatenated, wsclient->generated_key,
+         strlen((char *)wsclient->generated_key));
+  memcpy(concatenated + strlen((char *)wsclient->generated_key), WS_GUID,
```

Comment:
Here the `strlen` is used two times, we should only do it once. (and also use the strnlen method)

Reply:
Ohh I've just noticed, it is called three times in this method. please extract it to a local variable.